File: src/main/java/de/tum/cit/aet/artemis/programming/dto/SubmissionProcessingDTO.java
package de.tum.cit.aet.artemis.programming.dto;

import java.io.Serial;
import java.io.Serializable;
import java.time.ZonedDateTime;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record SubmissionProcessingDTO(long exerciseId, long participationId, String commitHash, ZonedDateTime submissionDate, ZonedDateTime buildStartDate,
        ZonedDateTime estimatedCompletionDate) implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;
}

File: src/main/webapp/app/entities/programming/submission-processing-dto.ts
import dayjs from 'dayjs/esm';

export class SubmissionProcessingDTO {
    public exerciseId?: number;
    public participationId?: number;
    public commitHash?: string;
    public submissionDate?: dayjs.Dayjs;
    public estimatedCompletionDate?: dayjs.Dayjs;
    public buildStartDate?: dayjs.Dayjs;
}
